photon-core/src/main/java/org/photonvision/jni/RknnObjectDetector.java

```diff
@@ -0,0 +1,116 @@
+package org.photonvision.jni;
+
+import java.lang.ref.Cleaner;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.opencv.core.Mat;
+import org.photonvision.common.configuration.NeuralNetworkModelManager;
+import org.photonvision.common.logging.LogGroup;
+import org.photonvision.common.logging.Logger;
+import org.photonvision.rknn.RknnJNI;
+import org.photonvision.vision.opencv.Releasable;
+import org.photonvision.vision.pipe.impl.NeuralNetworkPipeResult;
+
+/**
+ * A class to represent an object detector using the Rknn library.
+ *
+ * <p>TODO: When we start supporting more platforms, we should consider moving most of this code
+ * into a common "ObjectDetector" class to define the common interface for all object detectors.
+ */
+public class RknnObjectDetector implements Releasable {
+    /** logger for the RknnObjectDetector */
+    private static final Logger logger = new Logger(RknnDetectorJNI.class, LogGroup.General);
+
+    /** Cleaner instance to release the detector when it is no longer needed */
+    private final Cleaner cleaner = Cleaner.create();
+
+    /** Pointer to the native object */
+    private final long objPointer;
+
+    /** Model configuration */
+    private final NeuralNetworkModelManager.Model model;
+
+    /** Returns the model used by the detector. */
+    public NeuralNetworkModelManager.Model getModel() {
+        return model;
+    }
+
+    /** Atomic boolean to ensure that the detector is only released _once_. */
+    private AtomicBoolean released = new AtomicBoolean(false);
+
+    /**
+     * Creates a new RknnObjectDetector from the given model.
+     *
+     * @param model The model to create the detector from.
+     */
+    public RknnObjectDetector(NeuralNetworkModelManager.Model model) {
+        this.model = model;
+
+        // Create the detector
+        objPointer =
+                RknnJNI.create(model.modelFile.getPath(), model.labels.size(), model.version.ordinal(), -1);
+        if (objPointer <= 0) {
+            throw new RuntimeException(
+                    "Failed to create detector from path " + model.modelFile.getPath());
+        }
+
+        logger.debug("Created detector for model " + model.modelFile.getName());
+
+        // Register the cleaner to release the detector when it goes out of scope
+        cleaner.register(this, this::release);
+
+        // Set the detector to be released when the JVM exits
+        Runtime.getRuntime().addShutdownHook(new Thread(this::release));
```

Comment:
Out of scope: This is from back when the RGA resize was randomly causing the whole opi to lock up and v4l to crash horribly and terribly. We should restructure our thread model to make this not a thing anymore instead.

Reply:
Like you said I carried this over from the old implementation. 

Are you saying it's unnecessary now, or it could be made unnecessary by fixing something?

Reply:
I believe it's currently unnecessary. Let's just cut a new issue ticket to track it and resolve this thread tho.

Reply:
I've removed it, you know more about the existing issue. Could you create that?